### DISCUSSION TITLE
Canada > Quebec data

### DISCUSSION THREAD

**Discussion by christbelanger** [electricitymaps/electricitymaps-contrib#3210] [General]
Hi,

We just published a data source for enegy sources for Quebec.

You can find the info about the dataset here : https://www.hydroquebec.com/documents-data/open-data/electricity-generation-quebec/

And the json file here : https://www.hydroquebec.com/data/documents-donnees/donnees-ouvertes/json/production.json 

Hope it's "parser bildable" :-)

Christophe

**Comment by ArkadNaisir** [comment]
Just happy someone found the data (I opened a ticket with Hydro-Quebec to access it)

Does the fact the data is in French is an issue? Would think not since France provides data to the map.

Merci Christophe

**Reply by christbelanger** [reply]
Since I'm not a dev, I think the language issue would be address with the parser.

The translation of the energy sources can be found in the "Energy generated from renewable sources" widget located on this page :  https://www.hydroquebec.com/about/our-energy.html

Cheers

**Comment by bouchecl** [comment]
Issue #3218 is open.

**Comment by madsnedergaard** [comment]
It's awesome, and thanks for sharing @christbelanger!
Several contributors have already found it (#3218) 😄 

And @pjakobsen is already working on building the parser 👏 💪

**Reply by christbelanger** [reply]
Excellent news !

Thanks @bouchecl for your contribution 👌 and @pjakobsen for the parser 🔥.

**Reply by pjakobsen** [reply]
Thanks for the info.  Still working on it... I can see pricing methods in other parsers, but can't get pricing from the Quebec open data.  Do you need pricing for the map?  

There may be other things that the data is missing, it remains to be seen I suppose.

**Reply by bouchecl** [reply]
Quebec has no organized market since all electricity generated in the area is purchased at set prices by Hydro-Québec through long-term PPAs or legislation (for the 165 TWh/year "Heritage Pool").

**Reply by pjakobsen** [reply]
OK, after a bit of work, here's where I'm at:

Energy mix data by hour can be found here https://www.hydroquebec.com/data/documents-donnees/donnees-ouvertes/json/production.json

Quebec exports electricity via many interconnects https://www.hydroquebec.com/clean-energy-provider/

We can tell from https://www.hydroquebec.com/clean-energy-provider/markets/ontario.html that Quebec does import electricity at times though in very small amounts, as far as we can tell.  However there is no live source for this data,

The parse requirements: "Contain the net flow of cross-border exchanges. Not required if the exchanges are insignificant for the electricity consumption in the zone."

The exchange data does not seem to be available from a Quebec source, hence the parser requirement cannot be fulfilled with the real time data Quebec currently has available.   

Any suggestions or ideas?

**Reply by bouchecl** [reply]
Quebec is interconnected with four zones: Ontario, New Brunswick, ISO New England and New York ISO.  These balancing areas release their cross-border exchanges. Maybe, you could use the values provided by CA_ON.py, CA_NB.py, US_NEISO.py and US_NY.py and invert the sign (a transit from CA_ON -> CA_QC trade would need to be negative in QC and positive in ON)?

**Reply by ArkadNaisir** [reply]
Isn't connected to Labrador via Churchill Fall dam?

**Reply by bouchecl** [reply]
Yes. However, there is no publicly available data for on flows coming from that plant. Similarly, the CA_NS.py parser disregards the Maritime Link (a 500 MW HVDC link) inferface linking Nova Scotia and Newfoundland.

**Comment by pjakobsen** [comment]
@bouchecl with regards to your comment above 

> Quebec has no organized market since all electricity generated in the area is purchased at set prices by Hydro-Québec through long-term PPAs or legislation (for the 165 TWh/year "Heritage Pool").

What are the implications for implementing a `fetch_price` method for the Quebec CA-QC parser?   Should I look for a pricing scheme, or just leave out the method?

**Reply by christbelanger** [reply]
Hi @pjakobsen, I'll just leave it blank as there is no "wholesale" electricity price in Quebec.

The Quebec electricity market is open to competition, but there is no spot energy market.

**Comment by madsnedergaard** [comment]
Hey @christbelanger .
The page about open data (https://www.hydroquebec.com/documents-data/open-data/electricity-generation-quebec/) does not say anything about how the dates - but I assume they are in local timezone (GMT-4), is that correct? :)

**Reply by christbelanger** [reply]
Hi @madsnedergaard, 

That's right, UTC/GTM -4 !

**Reply by pjakobsen** [reply]
Ah, I see now the connections.   Bonjour @christbelanger  et merci ....for replying to my web form request at HydroQuebec.  Looking forward to seeing your open data work on the map.   👍

**Reply by bouchecl** [reply]
Yes it is GMT-4:00 in the summer (EDT); GMT -5:00 in the winter (EST). If the America/Montreal time zone is not available, use America/New York or America/Toronto.
